gematria/datasets/convert_bhive_to_llvm_exegesis_input.cc

```diff
@@ -38,8 +42,33 @@ constexpr std::string_view kMemNamePrefix = "MEM";
 
 ABSL_FLAG(std::string, bhive_csv, "", "Filename of the input BHive CSV file");
 ABSL_FLAG(
-    std::string, output_dir, "",
+    std::string, asm_output_dir, "",
     "Directory containing output files that can be executed by llvm-exegesis");
+ABSL_FLAG(std::string, json_output_dir, "",
+          "Directory containing JSON output files");
+ABSL_FLAG(
+    unsigned, json_split_count, std::numeric_limits<unsigned>::max(),
+    "The number of annotated basic blocks to include in a single JSON file");
+
+bool WriteJsonFile(llvm::json::Array to_write, size_t json_file_number,
+                   std::string json_output_dir) {
+  llvm::Twine json_output_file_path = llvm::Twine(json_output_dir)
+                                          .concat("/")
+                                          .concat(llvm::Twine(json_file_number))
+                                          .concat(".json");
```

Comment:
I know this comes from the previous code, but still:

Ideally, we'd use `llvm::path::append()` here to automatically correctly handle separators, or at least we should be using `llvm::path::get_separator()` to get the right path separator.

Reply:
Switched to using `llvm::path::append()` along with `llvm::SmallString` here.